### ISSUE TITLE
Issue with Fill

### ISSUE THREAD

**Issue by anavalo** [greatfrontend/greatfrontend#104]
## Your GreatFrontEnd account email (optional)

tasos@moveo.ai 

## Problematic section

- [ ] Question Description
- [ ] Question Skeleton
- [x] Question Solution
- [ ] Test Cases (Missing/Incorrect/Unclear)

## Type of question

- [x] JavaScript
- [ ] User Interface
- [ ] System Design
- [ ] Quiz

## Describe the issue

The current implementation of the fill function sets the end parameter to length + 1 when end is greater than the array's length. This approach does not align with standard JavaScript behavior where the end index is typically not inclusive in array operations. Adjusting end to simply equal the array's length would be more consistent with standard JavaScript practices.

## Code you used to submit (where relevant)

```js
if (end > length) {
    end = length;
}
```

## Anything else?

I propose modifying the line `end = length + 1;` to simply `end = length;`. This change would make the function's behavior more aligned with common JavaScript array handling practices, where the end index is non-inclusive and would not alter the intended functionality of the function.

**Comment by o-nedashkivska** [comment]
You are right, but this `if` check can be skipped as mentioned in the issue: https://github.com/greatfrontend/greatfrontend/issues/153

**Comment by yangshun** [comment]
Thanks for the feedback and for helping improve GreatFrontEnd!

Both bounds are clamped correctly and the length+1 branch is gone.

We've verified this against the current repository and are closing the issue as resolved.
